coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/CoralSqlUnnestOperator.java

```diff
@@ -0,0 +1,114 @@
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlOperatorBinding;
+import org.apache.calcite.sql.SqlUnnestOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.type.ArraySqlType;
+import org.apache.calcite.sql.type.MapSqlType;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+
+/**
+ * This class is Coral's representation for Hive explode function.
+ * Explode enables unboxing an array or map of an operand.
+ * The function returns a row set of single column for the array operand, or
+ * a row set with two columns corresponding to (key, value) for
+ * map operand type. The array operand may be a primitive datatype, such as int, String.
+ * Or, it may also be an abstract datatype, such as struct.
+ */
+public class CoralSqlUnnestOperator extends SqlUnnestOperator {
+
+  // _withOrdinality represents whether output should contain an ORDINALITY column
+  final boolean _withOrdinality;
+  // _sqlDialect represents the target language's dialect. By default, it is set to spark
+  String _sqlDialect = "spark";
+  // _relDataType represents the datatype of the operand to be unboxed.
+  RelDataType _relDataType;
+
+  public static final String ARRAY_ELEMENT_COLUMN_NAME = "col";
+  public static final String ARRAY_ELEMENT_POS_NAME = "pos";
+
+  public CoralSqlUnnestOperator(boolean withOrdinality, RelDataType relDataType) {
+    // keep the same as base class 'UNNEST' operator
+    // Hive has a separate 'posexplode' function for ordinality
+    super(withOrdinality);
+
+    _withOrdinality = withOrdinality;
+    _relDataType = relDataType;
+  }
+
+  public void setSqlDialect(String sqlDialect) {
+    _sqlDialect = sqlDialect;
+  }
+
+  public RelDataType getRelDataType() {
+    return _relDataType;
+  }
+
+  @Override
+  public boolean checkOperandTypes(SqlCallBinding callBinding, boolean throwOnFailure) {
+    RelDataType operandType = callBinding.getOperandType(0);
+    if (withOrdinality) {
+      return operandType instanceof ArraySqlType;
+    }
+    return operandType instanceof ArraySqlType || operandType instanceof MapSqlType;
+  }
+
+  @Override
+  public SqlOperandCountRange getOperandCountRange() {
+    return SqlOperandCountRanges.of(1);
+  }
+
+  @Override
+  public RelDataType inferReturnType(SqlOperatorBinding opBinding) {
+    RelDataType operandType = opBinding.getOperandType(0);
+    final RelDataTypeFactory.Builder builder = opBinding.getTypeFactory().builder();
+
+    if (withOrdinality) {
+      builder.add(ARRAY_ELEMENT_COLUMN_NAME, operandType.getComponentType());
+      builder.add(ARRAY_ELEMENT_POS_NAME, SqlTypeName.INTEGER);
```

Comment:
1. Calcite uses :
```
builder.add(SqlUtil.deriveAliasFromOrdinal(operand), type.getComponentType());
```
 to append the first column. The util `deriveAliasFromOrdinal(..)` is going to return `EXPR$0` as the column name. Historically we have been using "col", since Hive always uses this name for the unnested column. We could either use the parent implementation and change the name somewhere else, or we can go with the current simplified approach. 

2. As per legacy behavior, whenever the Ordinality column was needed, we would expose that with the name "pos". Thats why we don't use the default "ORDINALITY_COLUMN_NAME" (which translates to "ORDINALITY").

Reply:
Marking it resolved. Please reopen if there are any further followup questions.